pom.xml

```diff
@@ -20,7 +20,7 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
-    <java.version>1.8</java.version>
+    <java.version>17</java.version>
```

Comment:
We cannot require users to use 17

Reply:
Yeah, Like I already comment on this PR, I suggest this commit as a new version to separate support long term for Java 11, 17,... 
- The previous commit f10b510 also support completely to generate record file in Java 8. But we cannot use record in Java 8, so I want something like "If it is created, it must be usable".

Reply:
I will revert this commit, if my PR can be merged, we can discuss this ideal in another PR

Reply:
I had already reverted this commit, please review this PR again @JakeWharton